### ISSUE TITLE
'Type error: No overload matches this call' when building project in Vercel

### ISSUE THREAD

**Issue by ubirajaramneto** [nextauthjs/next-auth#7830]
### Environment

 System:
    OS: Windows 10 10.0.22621
    CPU: (20) x64 12th Gen Intel(R) Core(TM) i7-12700H
    Memory: 16.78 GB / 31.69 GB
  Binaries:
    Node: 18.16.0 - C:\Program Files\nodejs\node.EXE
    npm: 9.7.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1848.0), Chromium (114.0.1823.51)
    Internet Explorer: 11.0.22621.1

### Reproduction URL

Working on it...

### Describe the issue

When trying to deploy a new version in vercel, the build fails with the following stack trace in the vercel deployment logs:

![image](https://github.com/nextauthjs/next-auth/assets/5034480/145df581-8587-47c5-9fac-d260aba9f3ef)

If you visit the offending file in your local node_modules folder you will see the type error in the editor:

![image](https://github.com/nextauthjs/next-auth/assets/5034480/76513907-eb3a-4f4d-b0c8-a5575ef03407)

I am using NX.dev monorepo framework. Here are some of the dependencies for the project:

```json
"devDependencies": {
    "@nestjs/schematics": "^9.1.0",
    "@nestjs/testing": "^9.1.1",
    "@nx/cypress": "16.3.2",
    "@nx/eslint-plugin": "16.3.2",
    "@nx/jest": "16.3.2",
    "@nx/js": "16.3.2",
    "@nx/linter": "16.3.2",
    "@nx/nest": "16.3.2",
    "@nx/next": "^16.3.2",
    "@nx/node": "16.3.2",
    "@nx/react": "16.3.2",
    "@nx/webpack": "16.3.2",
    "@nx/workspace": "16.3.2",
    "@tailwindcss/typography": "^0.5.9",
    "@testing-library/react": "14.0.0",
    "@types/cookie": "^0.5.1",
    "@types/jest": "^29.4.0",
    "@types/node": "18.14.2",
    "@types/react": "18.2.8",
    "@types/react-dom": "18.0.11",
    "@typescript-eslint/eslint-plugin": "^5.58.0",
    "@typescript-eslint/parser": "^5.58.0",
    "autoprefixer": "10.4.13",
    "babel-jest": "^29.4.1",
    "cypress": "^12.11.0",
    "daisyui": "^3.1.0",
    "eslint": "~8.15.0",
    "eslint-config-next": "13.4.1",
    "eslint-config-prettier": "8.1.0",
    "eslint-plugin-cypress": "^2.10.3",
    "eslint-plugin-import": "2.27.5",
    "eslint-plugin-jsx-a11y": "6.7.1",
    "eslint-plugin-react": "7.32.2",
    "eslint-plugin-react-hooks": "4.6.0",
    "jest": "^29.4.1",
    "jest-environment-jsdom": "^29.4.1",
    "jest-environment-node": "^29.4.1",
    "nx": "16.3.2",
    "postcss": "8.4.21",
    "prettier": "^2.6.2",
    "tailwindcss": "3.2.7",
    "ts-jest": "^29.1.0",
    "ts-node": "10.9.1",
    "typescript": "~5.1.3"
  },
  "dependencies": {
    "@nestjs/common": "^9.1.1",
    "@nestjs/core": "^9.1.1",
    "@nestjs/platform-express": "^9.1.1",
    "axios": "^1.0.0",
    "dotenv": "^16.1.4",
    "encoding": "^0.1.13",
    "next": "13.4.4",
    "next-auth": "^4.22.1",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-icons": "^4.9.0",
    "reflect-metadata": "^0.1.13",
    "rxjs": "^7.8.0",
    "tslib": "^2.3.0"
  }
```

I am using AWS Cognito provider but I don't think this is related to the current setup that I have, as it does work properly without any issues in the current dev server NX.dev provides.

### How to reproduce

1 - Setup a new Next.js project using version 13.4.4 and NextAuth version 4.22.1 (typescript version is 5.1.3, not sure if this is related)
2 - Try to deploy to vercel
3 - The project should fail to build

### Expected behavior

The project should build

**Comment by ubirajaramneto** [comment]
A workaround I am using right now is to add the following to the next.config.js file:

```javascript
typescript: {
    ignoreBuildErrors: true,
}
```

**Comment by ubirajaramneto** [comment]
Sorry, after reading the source code from both next-auth and my project, I believe I found that I was importing maybe an obsolete or outdated type, which was causing the build to break. I refactored the code to remove that type and the problem disappeared. Just for context, this was the offending import that was breaking my build.

![image](https://github.com/nextauthjs/next-auth/assets/5034480/bf6ef1e7-d909-4af2-b48f-a46642fd4d94)

It was the `Session` type from `next-auth/src`
